src/backend/src/services/work-packages.services.ts

```diff
@@ -576,4 +578,29 @@ export default class WorkPackagesService {
       }
     });
   }
+
+  static async getBlockingWorkPackages(wbsNum: WbsNumber): Promise<WorkPackage[]> {
+    const { carNumber, projectNumber, workPackageNumber } = wbsNum;
+
+    if (workPackageNumber === 0) throw new HttpException(400, `${wbsPipe(wbsNum)} is not a valid work package WBS!`);
+
+    const workPackage = await prisma.work_Package.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      ...workPackageQueryArgs
+    });
+
+    if (!workPackage) throw new NotFoundException('Work Package', wbsPipe(wbsNum));
+
+    const blockingWbsElementIds = await getBlockingWbsElementIds(workPackage);
```

Comment:
these two functions are so complicated. i can get this in one prisma call. lmk if you need a hint (i am 90% sure my way works but who knows)